lib/cooked_post_processor.rb

```diff
@@ -60,16 +60,8 @@ def grant_badges
   def keep_reverse_index_up_to_date
     upload_ids = Set.new
 
-    @doc.css("a[href]").each do |a|
-      href = a["href"].to_s
-      if upload = Upload.get_from_url(href)
-        upload_ids << upload.id
-      end
-    end
-
-    @doc.css("img[src]").each do |img|
-      src = img["src"].to_s
-      if upload = Upload.get_from_url(src)
+    @doc.css("a/@href", "img/@src").each do |media|
```

Comment:
This change is a refactor and I don't see how it is changing the original behavior. But you're right that the other change would force us to check all other URLs. :thinking: We can match it against the current `Discourse.base_url`. I also noticed that we can actually just do a `Upload.where(url: [<urls>])` instead of executing many individual `find_by`s here